src/scaffoldmaker/meshtypes/meshtype_3d_bone1.py

```diff
@@ -0,0 +1,461 @@
+"""
+Generates a solid bone using a ShieldMesh of all cube elements,
+ with variable numbers of elements in major, minor, shell and axial directions.
+"""
+
+from __future__ import division
+
+import copy
+
+from cmlibs.utils.zinc.field import findOrCreateFieldCoordinates
+from cmlibs.utils.zinc.finiteelement import getMaximumNodeIdentifier, getMaximumElementIdentifier
+from cmlibs.zinc.field import Field
+from cmlibs.zinc.node import Node
+from scaffoldmaker.meshtypes.meshtype_1d_path1 import MeshType_1d_path1
+from scaffoldmaker.meshtypes.scaffold_base import Scaffold_base
+from scaffoldmaker.scaffoldpackage import ScaffoldPackage
+from scaffoldmaker.utils import vector
+from scaffoldmaker.utils.cylindermesh import CylinderMesh, CylinderShape, CylinderEnds, CylinderCentralPath
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.shieldmesh import ShieldMesh3D
+from scaffoldmaker.utils.spheremesh import SphereMesh, SphereShape
+from scaffoldmaker.utils.zinc_utils import exnode_string_from_nodeset_field_parameters
+from scaffoldmaker.utils.derivativemoothing import DerivativeSmoothing
+from scaffoldmaker.annotation.annotationgroup import AnnotationGroup, findOrCreateAnnotationGroupForTerm
+
+
+class MeshType_3d_bone1 (Scaffold_base):
+    """
+Generates a solid cylinder using a ShieldMesh of all cube elements,
```

Comment:
Ident to match indentation of start of docstring (""")

Reply:
Done